### ISSUE TITLE
BlockVariationTransforms: Improve clarity and labeling

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59898]
### Description

The `BlockVariationTransforms` component of the block editor is used to show variations of a block, for example for the Group block. It conditionally renders 3 different UIs, depending on some conditions:

- By default, it renders a radio buttons group labeled 'Transform to variation'.
- If the variations are more than 5, it renders icon buttons within a fieldset with a visually hidden legend 'Transform to variation'.
- If the variation icons are not unique, it renders a button labeled 'Transform to variation' with a dropdown menu labeled 'Transform to variation'.

Screenshot:

![BlockVariationTransforms](https://github.com/WordPress/gutenberg/assets/1682452/4b7777be-1e8a-4e5b-92bb-c7b6573f3939)

All these three UIs can be improved fo rbetter clarity and labeling.

## Radio group

![radiogroup](https://github.com/WordPress/gutenberg/assets/1682452/d9ddd275-665a-4c61-a93f-2164ac82ba7d)

- As a user, looking at this Ui, I don't understand what it is about. There is no visible text to explain these are variations of the block. I only see four icons that don't even look like buttons. These four controls are wrapped within a `role=radiogroup` with an aria-label 'Transform to variation'. That'e the equivalent of a native fieldset `<legend>` element. The hidden group name is less than ideal. It should be visible to clearly indicate what these controls are. Worth also noting that all the other settings sections do have a visible title.
- Under the hood, these four buttons have a `role=radio'. They are the equivalent of native radio buttons. In fact, they also behave like radio buttons where keyboard selection works with arrow keys. However, the keyboard interaction isn't predictable because the visuals doesn't really clarify these work like radio buttons.
- The labeling of the radio buttons is less than ideal. Each label _changes dynamically_ depending on the checked state. For example, 'Transform to Row' changes to only 'Row' when checked. Labels provide the accessible name of a control and they should never change dynamically/ To illustrate visually, in an equivalent group of native radio buttons this would be a weird behavior also visually, where the labels change depending on the state. Not the best way to use labels. Screenshot:

![radiogroup html](https://github.com/WordPress/gutenberg/assets/1682452/56b211cb-dea9-4cf2-946a-e9c20927b420)

## Icon buttons

- As above, there's no visible text to clarify what these buttons are about.
- The buttons use both a label prop and an aria-label and they mismatch. [Looking at the code](https://github.com/WordPress/gutenberg/blob/70cc05d89e069af45a50a936368fd420b1c8208f/packages/block-editor/src/components/block-variation-transforms/index.js#L41-L51), this seems intentional but it's less than ideal. For example, while the tooltip text shows 'Transform to Row', the actual accessible name is `aria-label="Row"`. A mismatch between visible label and accessible name is a barrier for assistive technology users, for example sighted screen reader users or speech recognition software users.

## Dropdown menu 

- This is the only case where some visible text clarifies what this control is about: 'Transform to variation'. However, the toggle button also shows a tooltip that just repeats the button visible text:

![Screenshot 2024-03-15 at 10 26 16](https://github.com/WordPress/gutenberg/assets/1682452/c978f60a-a025-46c2-bfb1-c386f307510b)

- In the dropdown menu, each menu item is a button. These buttons also show the variation description. However, the description is placed within the button text so that it contributes to compute the button accessible name, which is less than ideal. A button text that says, for example `Group Gather blocks in a container.` is just confusing. Accessible name and description should always be separated. This appears to be a more general problem with the `MenuItem` component, where the prop `info` just adds text to the button content. To illustrate visually, this would be the equivalent of a native button element with some really confusing text like in this example screenshot:

![Screenshot 2024-03-15 at 10 42 55](https://github.com/WordPress/gutenberg/assets/1682452/137af540-d64e-4ba1-a737-e28c96a7ff00)

(literally with no space between Group and Gather)

### Step-by-step reproduction instructions

- Go to the site editor and select a Group block.
- Observe the variations in the block inspector.
- Observe all the issues pointed out in this issue description above.
- Alter the [Group block variations](https://github.com/WordPress/gutenberg/blob/b6c37f383070d35af77d1aab9b9ef01df33c6d9d/packages/block-library/src/group/variations.js) and add a few more variations so that they are more than 5. Build and refresh the editor.
- Observe the variations are now icon buttons.
- Observe all the issues pointed out in this issue description above.
- Alter again the variations and make at least two of them use the same icon. Build and refresh the editor.
- Observe the variations are now a dropdown menu.
- Observe all the issues pointed out in this issue description above.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
While working on this, I noticed a visual issue that'd be nice to address. When the variations are rendered with a dropdown menu, the gap between the menu toggle button and the menu is too large. This isn't the expected gap and hte menu feels too disconnected from its toggle:

![menu gap](https://github.com/WordPress/gutenberg/assets/1682452/20b25f9c-ffb9-4aab-a97d-e51ffb746659)

I think this needs to be fixed by making sure all three components that render the variations are placed within the same wrapper that uses the same CSS class and padding.

**Comment by afercia** [comment]
Noting that even after recent changes from https://github.com/WordPress/gutenberg/pull/60220 the 'selected' and 'focused' states are still based on color only. As pointed out several tims in the history of this project, color only is not sufficient. The states must be distinguished with a different _shape_. 

For users with color perception impairments or environments where there's no color e.g. Windows High Contrast more, the two outlines are identical.

Screenshot sfrom latest trunk:

![Screenshot 2024-05-29 at 13 52 04](https://github.com/WordPress/gutenberg/assets/1682452/e4d4f1b9-1b5f-4ac1-96f1-78cb4df5f3e0)

![Screenshot 2024-05-29 at 13 52 38](https://github.com/WordPress/gutenberg/assets/1682452/f8c2af40-e393-4322-8506-3913c41e68f2)
